src/models/MultiTeam.php

```diff
@@ -35,22 +35,31 @@ class MultiTeam extends Team {
       $teams = await self::genTeamArrayFromDB('SELECT * FROM teams');
       foreach ($teams->items() as $team) {
         $all_teams->add(
-          Pair {intval($team->get("id")), Team::teamFromRow($team)},
+          Pair {intval($team->get('id')), Team::teamFromRow($team)},
         );
       }
       self::setMCRecords('ALL_TEAMS', $all_teams);
+      return $all_teams;
+    } else {
+      invariant(
+        ($mc_result !== null && $mc_result instanceof Map),
```

Comment:
I believe `$mc_result instanceof Map` should take care of the `$mc_result !== null` case, but you can double check that. Same for any other place where you do this.